lib/msal-browser/src/operatingcontext/BaseOperatingContext.ts

```diff
@@ -24,20 +29,87 @@ export abstract class BaseOperatingContext {
     protected available: boolean;
     protected browserEnvironment: boolean;
 
+    protected static LOGGER_CALLBACKS: Map<number, ILoggerCallback> = new Map([
+        [
+            LogLevel.Error,
+            // eslint-disable-next-line no-console
+            (level: LogLevel, message: string) => console.error(message),
+        ],
+        [
+            LogLevel.Warning,
+            // eslint-disable-next-line no-console
+            (level: LogLevel, message: string) => console.warn(message),
+        ],
+        [
+            LogLevel.Info,
+            // eslint-disable-next-line no-console
+            (level: LogLevel, message: string) => console.info(message),
+        ],
+        [
+            LogLevel.Verbose,
+            // eslint-disable-next-line no-console
+            (level: LogLevel, message: string) => console.debug(message),
+        ],
+        [
+            LogLevel.Trace,
+            // eslint-disable-next-line no-console
+            (level: LogLevel, message: string) => console.trace(message),
+        ],
+    ]);
+
     constructor(config: Configuration) {
         /*
          * If loaded in an environment where window is not available,
          * set internal flag to false so that further requests fail.
          * This is to support server-side rendering environments.
          */
         this.browserEnvironment = typeof window !== "undefined";
-
         this.config = buildConfiguration(config, this.browserEnvironment);
-        this.logger = new Logger(
-            this.config.system.loggerOptions,
-            name,
-            version
-        );
+
+        let sessionStorage: Storage | undefined;
+        try {
+            sessionStorage = window[BrowserCacheLocation.SessionStorage];
+            // Mute errors if it's a non-browser environment or cookies are blocked.
+        } catch (e) {}
+
+        const logLevelKey = sessionStorage?.getItem(LOG_LEVEL_CACHE_KEY);
+        const piiLoggingEnabled =
+            sessionStorage?.getItem(LOG_PII_CACHE_KEY)?.toLowerCase() ===
+            "true";
+
+        if (logLevelKey || piiLoggingEnabled) {
+            const logLevel =
+                logLevelKey && Object.keys(LogLevel).includes(logLevelKey)
+                    ? LogLevel[logLevelKey]
+                    : undefined;
+            const loggerCallback =
+                logLevelKey &&
+                BaseOperatingContext.LOGGER_CALLBACKS.has(LogLevel[logLevelKey])
+                    ? BaseOperatingContext.LOGGER_CALLBACKS.get(
+                          LogLevel[logLevelKey]
+                      )
+                    : undefined;
+            const loggerOptions = this.config.system.loggerOptions;
+
+            this.logger = new Logger(
+                {
+                    correlationId: loggerOptions.correlationId,
+                    loggerCallback:
+                        loggerCallback || loggerOptions.loggerCallback,
```

Comment:
Since we OR with the provided config values here the else condition below is likely redundant. Can we simplify?

Reply:
Given that this PR introduces two flags, customer may enable `msal.browser.log.pii` only, leaving `msal.browser.log.level` unset. In that case `loggerCallback` will be initialized as `undefined`, making us to fall back to the logger callback provided in the app config.

Reply:
Right understood. I'm suggesting that the else condition below can be made redundant if you refactor this a bit.